app/assets/v2/js/ptokens/ptokens.js

```diff
@@ -0,0 +1,88 @@
+const TX_STATUS_PENDING = 'pending';
+const TX_STATUS_SUCCESS = 'success';
+const TX_STATUS_ERROR = 'error';
+const TX_STATUS_UNKNOWN = 'unknown';
+const TX_STATUS_DROPPED = 'dropped';
+const redemption_states = [ 'request', 'accepted', 'denied', 'completed' ];
+
+function get_ptokens() {
+  return fetchData('/tokens/', 'GET');
+}
+
+function get_ptoken_redemptions(tokenId, state) {
+  if (redemption_states.indexOf(state)) {
+    return fetchData(`/tokens/${tokenId}/redemptions?state=${state}`, 'GET');
+  }
+
+  return fetchData('/tokens/', 'GET');
+}
+
+function create_ptoken(name, symbol, address, value, minted, owner_address, txId, web3_created) {
+  return fetchData('/tokens/', 'POST', {
+    'token_name': name,
+    'token_symbol': symbol,
+    'token_address': address,
+    'token_owner_address': owner_address,
+    'network': network,
+    'tx_status': TX_STATUS_PENDING,
+    'txid': txId,
+    'total_minted': minted,
+    'value': value,
+    'web3_created': web3_created
+  });
+}
+
+function mint_tokens(tokenId, amount) {
```

Comment:
Should there be more logic here, @coopahtroopanew? Any front end limits you want to put on people, like can only create 50 new tokens at a time, or only once every so often can more be minted? Here's probably best place for that.

Reply:
I think we're good to just let people run wild with it. That's basically what we landed on and worst-case scenario if someone mints way too much we can probably just let them burn them. Thoughts?